Multisig/UI/ClaimToken/Select Guardian/GuardianListViewController.swift

```diff
@@ -88,32 +92,34 @@ class GuardianListViewController: LoadableViewController {
 
     override func reloadData() {
         super.reloadData()
-        currentDataTask?.cancel()
-        currentDataTask = App.shared.claimingService.asyncGuardians() { [weak self] result in
+
+        controller.guardians(for: safe.addressValue) { [weak self] result in
             guard let `self` = self else { return }
-            switch result {
-            case .failure(let error):
-                DispatchQueue.main.async { [weak self] in
-                    guard let `self` = self else { return }
-                    // ignore cancellation error due to cancelling the
-                    // currently running task. Otherwise user will see
-                    // meaningless message.
-                    if (error as NSError).code == URLError.cancelled.rawValue &&
-                        (error as NSError).domain == NSURLErrorDomain {
-                        return
-                    }
-
-                    self.onError(GSError.error(description: "Failed to load guardians", error: error))
-                    self.onReloaded?()
-                }
-            case .success(let results):
-                DispatchQueue.main.async { [weak self] in
-                    guard let `self` = self else { return }
+            do {
+                let data = try result.get()
+                self.selectedDelegate = data.delegate
+
+                // TODO: move shuffling to the store
```

Comment:
find a way how to keep shuffled order after re-opening this screen within the same flow

Reply:
Decided to keep it like it is, it seems to work OK.